packages/providence-analytics/src/cli/cli.js

```diff
@@ -157,39 +161,49 @@ async function cli({ cwd, addProjectDependencyPaths } = {}) {
       '-t, --search-target-paths [targets]',
       `path(s) to project(s) on which analysis/querying should take place. Requires
     a list of comma seperated values relative to project root`,
-      v => pathsArrayFromCs(v, cwd),
+      v => cliHelpers.pathsArrayFromCs(v, cwd),
       targetDefault(),
     )
     .option(
       '-r, --reference-paths [references]',
       `path(s) to project(s) which serve as a reference (applicable for certain analyzers like
     'match-imports'). Requires a list of comma seperated values relative to
     project root (like 'node_modules/lion-based-ui, node_modules/lion-based-ui-labs').`,
-      v => pathsArrayFromCs(v, cwd),
+      v => cliHelpers.pathsArrayFromCs(v, cwd),
       InputDataService.referenceProjectPaths,
     )
     .option('-w, --whitelist [whitelist]', `whitelisted paths, like './src, ./packages/*'`, v =>
-      pathsArrayFromCs(v, cwd),
+      cliHelpers.pathsArrayFromCs(v, cwd),
     )
     .option(
       '--whitelist-reference [whitelist-reference]',
       `whitelisted paths for reference, like './src, ./packages/*'`,
-      v => pathsArrayFromCs(v, cwd),
+      v => cliHelpers.pathsArrayFromCs(v, cwd),
     )
     .option(
       '--search-target-collection [collection-name]',
       `path(s) to project(s) which serve as a reference (applicable for certain analyzers like
     'match-imports'). Should be a collection defined in providence.conf.js as paths relative to
     project root.`,
-      v => pathsArrayFromCollectionName(v, 'search-target', externalConfig),
+      v => cliHelpers.pathsArrayFromCollectionName(v, 'search-target', externalConfig),
     )
     .option(
       '--reference-collection [collection-name]',
       `path(s) to project(s) on which analysis/querying should take place. Should be a collection
     defined in providence.conf.js as paths relative to project root.`,
-      v => pathsArrayFromCollectionName(v, 'reference', externalConfig),
+      v => cliHelpers.pathsArrayFromCollectionName(v, 'reference', externalConfig),
     )
-    .option('--write-log-file', `Writes all logs to 'providence.log' file`);
+    .option('--write-log-file', `Writes all logs to 'providence.log' file`)
+    .option(
+      '--include-target-deps',
+      `For all search targets, will include all its dependencies
+    (node_modules and bower_components). When --target-deps-filter is applied, a subset
+    will be applied that matches the filter condition`,
+    )
+    .option(
+      '--target-deps-filter [target-deps-filter]',
```

Comment:
Do we need both `include-target-deps` and `target-deps-filter` ?

Proposal:
`--target-dependencies` --> enum `true | false | regex | glob` ?

Reply:
Yeah, was also thinking about that, but don't think commander really supports that. Or do you mean the string 'true' like `--target-dependencies 'true'`?

Reply:
Yeah exactly, a string enum

Reply:
Could also make it more clear by saying `all | none | <regex> | <glob>` (none = default) so people don't get confused thinking it's a boolean flag.

Reply:
Yeah, the problem is also a bit that a package could theoretically be called 'true', 'false' etc.

But I like the idea of globs, makes it more consistent with the rest (difference though is that we are dealing with package names instead of paths here). It would also be possible to discriminate between globs and regexes by requiring the latter to be enclosed by '/'. 

So allow `--target-dependencies /^my-prefix.*/` or `--target-dependencies glob*` ...?

Reply:
I think if you have globs you probably don't need regex right? Since globs pretty much covers "regex for file paths". And if you have a package called `all` or `none` all you gotta do is make the path to it non-absolute `all/` or `./all`, should be okay I think :P

Reply:
The match is actually not based on path names, but on folder names.
We use `read-package-tree`: https://github.com/ing-bank/lion/blob/master/packages/providence-analytics/src/program/utils/read-package-tree-with-bower-support.js#L153 (an inline fork, because we need bower support).
This could possibly be rewritten later, but for now I would propose the regex notation

Reply:
FIxed 💪

Reply:
Nice, what is the plan for migrating to globs? Any timeline or open issue for it? I think it's good for now but for many people this will be rather unexpected to use a regex instead of glob for file pattern.